code/go/0chain.net/smartcontract/dbs/event/provider.go

```diff
@@ -68,15 +57,52 @@ func (p *Provider) BeforeCreate(tx *gorm.DB) (err error) {
 	return
 }
 
+func (edb *EventDb) updateProviderTotalStakes(providers []Provider, tablename string) error {
+	var ids []string
+	var stakes []int64
+	for _, m := range providers {
+		ids = append(ids, m.ID)
+		i, err := m.TotalStake.Int64()
+		if err != nil {
+			return err
+		}
+		stakes = append(stakes, i)
+	}
+
+	return CreateBuilder(tablename, "id", ids).
+		AddUpdate("total_stake", stakes).Exec(edb).Error
+}
+
+func (edb *EventDb) updateProvidersTotalUnStakes(providers []Provider, tablename string) error {
+	var ids []string
+	var unstakes []int64
+	for _, m := range providers {
+		ids = append(ids, m.ID)
+		i, err := m.TotalStake.Int64()
```

Comment:
It should be `m.UnstakeTotal`, we need tests to ensure the total unstakes is updated correctly

Reply:
we will add it in the next issue, we need to merge it asap